base/math.jl

```diff
@@ -339,6 +339,10 @@ minmax{T<:AbstractFloat}(x::T, y::T) =
            ifelse((y > x) | (signbit(x) > signbit(y)), (x,y), (y,x)))
 
 
+inttype(::Type{Float64}) = Int64
+inttype(::Type{Float32}) = Int32
+inttype(::Type{Float16}) = Int16
+exponent_max{T<:AbstractFloat}(::Type{T}) = Int(exponent_mask(T) >> significand_bits(T))
```

Comment:
Yeah we don't need `inttype` using your comment below.

Another case  where I think this function would be useful is for something like this
`intasfloat{T<:IEEEFloat}(::Type{T}, m::Integer) = reinterpret(T, (m % inttype(T)) << significand_bits(T))`
 
but personally I think it's best to not include it, until there's actually another function that would benefit.

for `exponent_max` I've just inlined it. It seems rather special case to add it as a another function in `float.jl`. However, if you have objections I don't mind to put it in float.jl or similar